src/v/iceberg/avro_decimal.h

```diff
@@ -0,0 +1,77 @@
+// Copyright 2024 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+#pragma once
+
+#include "bytes/iobuf.h"
+
+#include <absl/numeric/int128.h>
+
+#include <array>
+namespace iceberg {
+/**
+ * Converts a decimal into an array of bytes (big endian)
+ */
+
+inline std::array<uint8_t, 16> encode_avro_decimal(absl::int128 decimal) {
```

Comment:
Maybe our `bytes` type is fine because of SSO if that makes stuff easier...

Reply:
i think so, will change that